src/main/java/com/github/fge/jsonpatch/AddOperation.java

```diff
@@ -65,78 +65,62 @@
  *     [ 1, 2, 3 ]
  * </pre>
  */
-public final class AddOperation
-    extends PathValueOperation
-{
-    private static final ReferenceToken LAST_ARRAY_ELEMENT
-        = ReferenceToken.fromRaw("-");
+public final class AddOperation extends PathValueOperation {
+    public static final String LAST_ARRAY_ELEMENT_SYMBOL = "-";
 
     @JsonCreator
-    public AddOperation(@JsonProperty("path") final JsonPointer path,
-        @JsonProperty("value") final JsonNode value)
-    {
+    public AddOperation(@JsonProperty("path") final String path,
+                        @JsonProperty("value") final JsonNode value) {
         super("add", path, value);
     }
 
     @Override
-    public JsonNode apply(final JsonNode node)
-        throws JsonPatchException
-    {
+    public JsonNode apply(final JsonNode node) throws JsonPatchException {
         if (path.isEmpty())
             return value;
-
         /*
          * Check the parent node: it must exist and be a container (ie an array
          * or an object) for the add operation to work.
          */
-        final JsonNode parentNode = path.parent().path(node);
-        if (parentNode.isMissingNode())
-            throw new JsonPatchException(BUNDLE.getMessage(
-                "jsonPatch.noSuchParent"));
+        final int lastSlashIndex = path.lastIndexOf('/');
+        final String newNodeName = path.substring(lastSlashIndex + 1);
+        final String pathToParent = path.substring(0, lastSlashIndex);
+        final String jsonPath = JsonPathParser.tmfStringToJsonPath(pathToParent);
+        DocumentContext nodeContext = JsonPath.parse(node.deepCopy());
+
+        final JsonNode parentNode = nodeContext.read(jsonPath);
+        if (parentNode == null)
```

Comment:
I would put `{` `}` around, same applies to other one-liner ifs

Reply:
That should be done for all one liner ifs.

Reply:
Done